[Merged by Bors] - Add missing readmes (#6031)
leanprover-community/mathlib4: Archive/Imo/README.md

# IMO problems

We have a collection of solutions to IMO problems in mathlib, stored under `/Archive/Imo/`.

These are part of mathlib for three purposes:
* The [IMO Grand Challenge](https://imo-grand-challenge.github.io/) will need training data, and exemplars,
  and this is a reasonable place to collect Lean samples of IMO problems.

> * As with the rest of `/archive/`, we want to have high quality examples

Reviewer: ```suggestion
* As with the rest of `/Archive/`, we want to have high quality examples
```

> * As with the rest of `/Archive/`, we want to have high quality examples

Reviewer 2: ```suggestion
* As with the rest of `Archive/`, we want to have high quality examples
```
seems more correct to me (like `Mathlib/` in another file), because you'd `cd Archive/` rather than `cd /Archive/` when you're in mathlib's root directory. (4 occurrences)

* Reviewing may take some time
  (because reviewers are motivated more by wanting to teach you how to write good code,
  rather than personal enthusiasm for the material).
* It is essential to read the
  [style guide](https://leanprover-community.github.io/contribute/style.html)
  carefully, and try to follow it very closely.
* Please document your solution thoroughly.
  Remember everything in `/Archive/` should be an exemplar of good mathlib style.
  There must be a clear human readable account of the solution at the top of the file
  (ideally something that would score 7!) but don't stop there.
  Please write doc-strings on most of your lemmas, explaining what they are doing and why,
  and in any non-trivial proof include single line comments explaining the significant steps.
* Be nice to your reviewers, and responsive to their suggestions!
  You should expect that your first pull request will involve at least as much work
  after you open the pull request as before.
  Reviewers may ask you to factor out useful lemmas into mathlib,
  or to completely restructure your proof.
* Pay forward what you learn from reviewing:

>   there are often several [open pull requests about IMO problems](https://github.com/leanprover-community/mathlib/pulls?q=is%3Aopen+is%3Apr+label%3Aimo),

Reviewer: This points to mathlib 3 repository. I guess, we should list both for now.

Author: [](url)
```suggestion
  there are often several [open pull requests about IMO problems](https://github.com/leanprover-community/mathlib4/pulls?q=is%3Aopen+is%3Apr+label%3Aimo) (and in [lean 3](https://github.com/leanprover-community/mathlib4pulls?q=is%3Aopen+is%3Apr+label%3Aimo),
```

>   there are often several [open pull requests about IMO problems](https://github.com/leanprover-community/mathlib4/pulls?q=is%3Aopen+is%3Apr+label%3Aimo) (and in [lean 3](https://github.com/leanprover-community/mathlib4pulls?q=is%3Aopen+is%3Apr+label%3Aimo),

Reviewer 2: ```suggestion
  there are often several [open pull requests about IMO problems](https://github.com/leanprover-community/mathlib4/pulls?q=is%3Aopen+is%3Apr+label%3Aimo) (and in [Lean 3](https://github.com/leanprover-community/mathlib/pulls?q=is%3Aopen+is%3Apr+label%3Aimo),
```
